contracts/router/modules/pool/traderjoe/TraderJoeV21Module.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {ILBRouter} from "../../../interfaces/traderjoe/ILBRouter.sol";
+import {ILBPair} from "../../../interfaces/traderjoe/ILBPair.sol";
+import {IndexedToken, IPoolModule} from "../../../interfaces/IPoolModule.sol";
+
+import {TraderJoeModule} from "./TraderJoeModule.sol";
+
+contract TraderJoeV21Module is TraderJoeModule {
+    constructor(address _lbRouter) TraderJoeModule(_lbRouter) {}
+
+    function version() public view override returns (ILBRouter.Version) {
+        return ILBRouter.Version.V2_1;
+    }
+
+    function _binStep(address pool) internal view override returns (uint256) {
+        return uint256(ILBPair(pool).getBinStep());
+    }
+
+    /// @inheritdoc IPoolModule
+    function getPoolQuote(
+        address pool,
+        IndexedToken memory tokenFrom,
+        IndexedToken memory tokenTo,
+        uint256 amountIn,
+        bool probePaused
+    ) public view override returns (uint256 amountOut) {
+        address[] memory tokens = getPoolTokens(pool);
+        bool swapForY = (tokenTo.token == tokens[1]);
+        require(amountIn <= type(uint128).max, "amountIn > type(uint128).max");
+
+        (, uint128 _amountOut, ) = lbRouter.getSwapOut(ILBPair(pool), uint128(amountIn), swapForY);
+        amountOut = uint256(_amountOut);
```

Comment:
Again, explicit casting is not required when you're casting to a bigger-sized type:
```suggestion
        (, amountOut,) = lbRouter.getSwapOut(ILBPair(pool), uint128(amountIn), swapForY);
```

Reply:
We should not be ignoring returned value of `amountInLeft`. Non-zero value means that swap is not possible and we should return `amountOut = 0`.

https://docs.traderjoexyz.com/contracts/LBPair#getswapout

You could check this in the `TraderJoeV21ModuleAvaxTestFork` test by setting the swapped amount to `10**6`. This leads to non-zero quote for `USDT -> USDC` swap, while the swap itself fails.

Reply:
good catch. fixed in https://github.com/synapsecns/synapse-contracts/pull/287/commits/1de556f5d6e2a6cbef2938b8057f8d5283edf112